Plater.lua

```diff
@@ -8259,6 +8259,16 @@ end
 		end
 	end
 	
+	function Plater.GetBgPlayerRoleCache(unit)
+		local name = GetUnitName(unit, true)
+		if not BG_PLAYER_CACHE[name] then
+			Plater.UpdateBgPlayerRoleCache()
+		end
+		
+		local cache = BG_PLAYER_CACHE[name]

```

Comment:
Simple `return BG_PLAYER_CACHE[name]` suffices, no need for another local in this case.